pkg/router/functionHandler.go

```diff
@@ -220,13 +221,18 @@ func (roundTripper *RetryingRoundTripper) RoundTrip(req *http.Request) (*http.Re
 			req.URL.Scheme = roundTripper.serviceURL.Scheme
 			req.URL.Host = roundTripper.serviceURL.Host
 
-			// To keep the function run container simple, it
-			// doesn't do any routing.  In the future if we have
-			// multiple functions per container, we could use the
-			// function metadata here.
-			// leave the query string intact (req.URL.RawQuery)
-			req.URL.Path = "/"
-
+			functionNamespacedURL := "/" + fnMeta.Namespace + "/" + fnMeta.Name
```

Comment:
@RealHarshThakur Can you please comments on respective blocks ? We should highlight `why` part considering conditional flow here.

Reply:
Wait does infinite wait, if we are stuck in package build times would increase and may keep running for a longer duration. Probably need to have wait with some timeout. 
I can try this as a separate PR where a fix would be applicable across all tests.